emily_cron/app/services/deposit_processor.py

```diff
@@ -0,0 +1,155 @@
+import logging
+from datetime import datetime
+from itertools import chain
+from typing import Iterable
+
+from ..clients import PublicEmilyAPI, PrivateEmilyAPI, HiroAPI, MempoolAPI
+from ..models import (
+    DepositUpdate,
+    EnrichedDepositInfo,
+    RequestStatus,
+    BlockInfo,
+    DepositInfo,
+)
+from .. import settings
+
+logger = logging.getLogger(__name__)
+
+
+class DepositProcessor:
+    """Service for processing deposits."""
+
+    def process_expired_locktime(
+        self,
+        enriched_deposits: list[EnrichedDepositInfo],
+        bitcoin_chaintip: BlockInfo,
+        stacks_chaintip: BlockInfo,
+    ) -> list[DepositUpdate]:
+        """Process transactions with expired locktime.
+
+        Args:
+            enriched_deposits: List of enriched deposit information
+            bitcoin_chaintip: Current Bitcoin block info
+            stacks_chaintip: Current Stacks block info
+
+        Returns:
+            list[DepositUpdate]: List of deposit updates
+        """
+        updates = []
+
+        # Find transactions with expired locktime
+        locktime_expired_txs = [
+            tx
+            for tx in enriched_deposits
+            if tx.confirmed_height > 0  # Only process confirmed transactions
+            and bitcoin_chaintip.height
+            >= tx.confirmed_height
+            + tx.lock_time
+            + settings.MIN_BLOCK_CONFIRMATIONS  # Check if locktime has expired
+        ]
```

Comment:
This may not be enough, if we have a sweep tx from the signers in the mempool and no signers activity after that; the correct way to handle this case is a bit annoying (see how we check for withdrawals _safe to reject_ case).
If we erroneously mark a deposit as failed but the signers go ahead and (correctly) process the request, will the sidecar be able to update this request as confirmed (once minted)?